manifests/setting.pp

```diff
@@ -0,0 +1,52 @@
+define apt::setting (
+  $type,
+  $priority   = 50,
+  $ensure     = file,
+  $source     = undef,
+  $content    = undef,
+  $file_perms = {},
+) {
+
+  $_file = merge($::apt::file_defaults, $file_perms)
+
+  if $content and $source {
+    fail('apt::setting cannot have both content and source')
+  }
+
+  if !$content and !$source {
+    fail('apt::setting needs either of content or source')
+  }
+
+  validate_re($type, ['conf', 'pref', 'list'])
+  validate_re($ensure,  ['file', 'present', 'absent'])
+
+  unless is_integer($priority) {
+    fail('apt::setting priority must be an integer')
+  }
+
+  if $source {
+    validate_string($source)
+  }
+
+  if $content {
+    validate_string($content)
+  }
+
+  if $type == 'list' {
+    $_priority = ''
```

Comment:
Ok, cool. At some point there will be massive doc updates, I ... haven't been keeping up with that.